### PR TITLE
[VL] Improve package scripts

### PR DESCRIPTION
**Pull request by wForget**
## What changes were proposed in this pull request?

Improve package scripts:

+ Extract `build-thirdparty.sh` script
+ Set `PROMPT_ALWAYS_RESPOND=y` for package scripts
+ Pass package arguments to `buildbundle-veloxbe.sh`
+ Add build thirdparty for `package-vcpkg.sh`

## How was this patch tested?

Manual package test

### CODE REVIEW

**Comment by FelixYBW** [comment]
@weiting-chen Do we still need the package.sh if we switch to static build? I never used the .sh.

**Comment by wForget** [comment]
>  if we switch to static build?

@FelixYBW Does `static build` mean using `vcpkg`?

**Comment by FelixYBW** [comment]
> 
> @FelixYBW Does `static build` mean using `vcpkg`?

Yes.

**Comment by wForget** [comment]
> > @FelixYBW Does `static build` mean using `vcpkg`?
> 
> Yes.

Got it, thanks. I also prefer this packaging way. This PR also adds build thirdparty for `package-vcpkg.sh`. Could you help review it?

File: dev/package-vcpkg.sh [hunk 1]
```diff
@@ -13,5 +16,13 @@ if [ "$LINUX_OS" == "centos" ]; then
...
+# prepare vcpkg environments
 source ./dev/vcpkg/env.sh
-./dev/buildbundle-veloxbe.sh --build_tests=ON --build_benchmarks=ON --enable_s3=ON  --enable_hdfs=ON
+
+# build gluten with velox backend, prompt always respond y
+export PROMPT_ALWAYS_RESPOND=y
+./dev/buildbundle-veloxbe.sh --build_tests=ON --build_benchmarks=ON --enable_s3=ON  --enable_hdfs=ON "$@"
+
+# make thirdparty package
+./dev/build-thirdparty.sh
```
Comment (PHILO-HE) [inline dev/package-vcpkg.sh:28]: This script is for vcpkg build with all dependencies statically linked. No need to pack shared libs.

Comment (wForget) [reply dev/package-vcpkg.sh:28]: > This script is for vcpkg build with all dependencies statically linked. No need to pack shared libs.

Sorry, I didn't look at the document carefully, does it mean we no longer need the thirdparty lib jar when using `package-vcpkg.sh`? I will try it, thank you.

Comment (PHILO-HE) [reply dev/package-vcpkg.sh:28]: @wForget, yes, third-party jar is useless when vcpkg build is enabled.

Comment (wForget) [reply dev/package-vcpkg.sh:28]: Thank you, I will close this PR.

Comment (wForget) [reply dev/package-vcpkg.sh:28]: > Extract build-thirdparty.sh script
> Set PROMPT_ALWAYS_RESPOND=y for package scripts
> Pass package arguments to buildbundle-veloxbe.sh

Are these improvements reasonable? If so, I will continue working

Comment (PHILO-HE) [reply dev/package-vcpkg.sh:28]: @wForget , yes, you can just keep these changes. Thanks!

**Review by PHILO-HE — APPROVED** [review]
Thanks!

**Comment by PHILO-HE** [comment]
@wForget, the weekly build job fails, seems related to this pr. Could you help take a look?
https://github.com/apache/incubator-gluten/actions/runs/10238892265/job/28323839510

**Comment by wForget** [comment]
> @wForget, the weekly build job failed, seems related to this pr. Could you help take a look? https://github.com/apache/incubator-gluten/actions/runs/10238892265/job/28323839510

Sorry, this PR unexpectedly introduced the following code. I will delete it.

https://github.com/wForget/gluten/blob/671f874aa0557fc0a6632e65667dce9a25eafeea/dev/package.sh#L12-L18

**Pull request merged**